### PR TITLE
Fix builds

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Removed
- Symfony deprecations
```

### CODE REVIEW

File: tests/Controller/CategoryAdminControllerTest.php [hunk 1]
```diff
@@ -261,7 +261,7 @@ public function testTreeAction($context, array $categories): void
 
         if (false !== $context) {
             $contextMock = $this->getContextMock($context);
-            $this->request->query->set('context', $contextMock->getId());
+            $this->request->query->set('context', $contextMock->getId() ?? '');
```
Comment (VincentLanglet) [inline tests/Controller/CategoryAdminControllerTest.php]:
```suggestion
            $this->request->query->set('context', $context);
```
The id is the argument of `getContextMock`

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: tests/Controller/CategoryAdminControllerTest.php (hunk 1 above)
Comment (jordisala1991) [reply tests/Controller/CategoryAdminControllerTest.php]: Way better!

**Review by VincentLanglet — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Comment by VincentLanglet** [comment]
I just tried to do the same in SonataMedia and discovered some admin was using POST request
https://github.com/sonata-project/SonataMediaBundle/blob/4.x/src/Admin/BaseMediaAdmin.php#L130

Are you sure about the `query->get` thing @core23 @jordisala1991 ?

**Comment by jordisala1991** [comment]
> I just tried to do the same in SonataMedia and discovered some admin was using POST request
> 
> https://github.com/sonata-project/SonataMediaBundle/blob/4.x/src/Admin/BaseMediaAdmin.php#L130
> 
> 
> 
> Are you sure about the `query->get` thing @core23 @jordisala1991 ?

Not 100% sure tbh. About that post thing, it is used to know when the form is submitted AFAIK

**Comment by VincentLanglet** [comment]
> > I just tried to do the same in SonataMedia and discovered some admin was using POST request
> > https://github.com/sonata-project/SonataMediaBundle/blob/4.x/src/Admin/BaseMediaAdmin.php#L130
> > Are you sure about the `query->get` thing @core23 @jordisala1991 ?
> 
> Not 100% sure tbh. About that post thing, it is used to know when the form is submitted AFAIK

I'm not familiar enough with the request, but when POST request are made, persistent parameters are in the `$request->request` instead of `$request->query` no ? Doesn't it means we're loosing all the persistent parameters when doing a post request then with this PR ?

**Comment by VincentLanglet** [comment]
I don't use this bundle. @core23 I think you're using it ; I would recommend to try those changes before doing another release.

**Comment by core23** [comment]
> I don't use this bundle. @core23 I think you're using it ; I would recommend to try those changes before doing another release.

👍  I'll check this before the next release.

**Pull request merged**